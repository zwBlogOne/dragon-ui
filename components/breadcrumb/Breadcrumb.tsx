import React, { Component, cloneElement, ReactElement } from 'react';
import classnames from 'classnames';
import { PropsType } from './PropsType';

class Breadcrumb extends Component<PropsType, any> {
  static defaultProps = {
    prefixCls: 'ui-breadcrumb',
    separator: '/',
  };

  static Item;

  render() {
    const {
      className, separator, children, style,
    } = this.props;

    const cls = classnames({
      'ui-breadcrumb': true,
      [className!]: !!className,
    });

    // eslint-disable-next-line
    const items = React.Children.map(children, (element, index) => {
      return cloneElement(element as ReactElement<any>, {
        separator,
        key: index,
      });
    });

    return (
      <div className={cls} style={style}>
        {items}
      </div>
    );
  }
}

export default Breadcrumb;
