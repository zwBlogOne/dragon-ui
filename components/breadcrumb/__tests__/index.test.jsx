import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Breadcrumb from '../index';

describe('Breadcrumb', () => {
  it('renders basic Breadcrumb correctly', () => {
    const wrapper = render(
      <Breadcrumb>
        <Breadcrumb.Item>首页</Breadcrumb.Item>
        <Breadcrumb.Item>模块</Breadcrumb.Item>
        <Breadcrumb.Item href="https://t.zhongan.com">应用</Breadcrumb.Item>
      </Breadcrumb>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Breadcrumb with customized separator correctly', () => {
    const wrapper = render(
      <Breadcrumb separator=">">
        <Breadcrumb.Item>首页</Breadcrumb.Item>
        <Breadcrumb.Item>模块</Breadcrumb.Item>
        <Breadcrumb.Item href="https://t.zhongan.com">应用</Breadcrumb.Item>
      </Breadcrumb>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
