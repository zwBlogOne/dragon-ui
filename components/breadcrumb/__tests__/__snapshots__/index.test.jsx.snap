// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders Breadcrumb with customized separator correctly 1`] = `
<div
  class="ui-breadcrumb"
>
  <span
    class=""
  >
    <span
      class="ui-breadcrumb-link"
    >
      首页
    </span>
    <span
      class="ui-breadcrumb-separator"
    >
      &gt;
    </span>
  </span>
  <span
    class=""
  >
    <span
      class="ui-breadcrumb-link"
    >
      模块
    </span>
    <span
      class="ui-breadcrumb-separator"
    >
      &gt;
    </span>
  </span>
  <span
    class=""
  >
    <a
      class="ui-breadcrumb-link"
      href="https://t.zhongan.com"
    >
      应用
    </a>
    <span
      class="ui-breadcrumb-separator"
    >
      &gt;
    </span>
  </span>
</div>
`;

exports[`Breadcrumb renders basic Breadcrumb correctly 1`] = `
<div
  class="ui-breadcrumb"
>
  <span
    class=""
  >
    <span
      class="ui-breadcrumb-link"
    >
      首页
    </span>
    <span
      class="ui-breadcrumb-separator"
    >
      /
    </span>
  </span>
  <span
    class=""
  >
    <span
      class="ui-breadcrumb-link"
    >
      模块
    </span>
    <span
      class="ui-breadcrumb-separator"
    >
      /
    </span>
  </span>
  <span
    class=""
  >
    <a
      class="ui-breadcrumb-link"
      href="https://t.zhongan.com"
    >
      应用
    </a>
    <span
      class="ui-breadcrumb-separator"
    >
      /
    </span>
  </span>
</div>
`;
